File: mishaa/OrednaApp/client/src/app/_models/apartment.ts
import { Photo } from "./photo"

export interface Apartment {
    photoUrl: string;
    title: string;
    details: string;
    city: string;
    country: string;
    price: number;
    maxPeople: number;
    hasTV: boolean;
    hasWifi: boolean;
    isAvaliable: boolean;
    photos: Photo[];
    id: number;
}

File: mishaa/OrednaApp/API/Entities/Apartment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

[Table("Apartments")]
public class Apartment : BaseEntity
{
    public string? PhotoUrl { get; set; }
    
    public string Title { get; set; }
    public string? Details { get; set; }

    public string City { get; set; }
    public string Country { get; set; }
    
    public float Price { get; set; }
        
    public int MaxPeople { get; set; }
        
    public bool HasTV { get; set; }
    public bool HasWifi { get; set; }
    
    public bool IsAvaliable { get; set; }

    public AppUser? RentedByUser { get; set; }
    public int? RentedByUserId { get; set; }

    public int AppUserId { get; set; }
    public AppUser AppUser { get; set; }
    
    public List<Photo> Photos { get; set; } = new List<Photo>();
}